File: VB/WindowsApplication54/ReportExtension.vb
Imports System.IO
Imports DevExpress.XtraReports.Extensions
Imports DevExpress.XtraReports.UI

Public Class ReportExtension
    Inherits ReportStorageExtension
    Public Overrides Sub SetData(ByVal report As XtraReport, ByVal stream As Stream)
        report.SaveLayoutToXml(stream)
    End Sub
End Class

File: CS/WindowsApplication54/ReportExtension.cs
using System.IO;
using DevExpress.XtraReports.Extensions;
using DevExpress.XtraReports.UI;

class ReportExtension : ReportStorageExtension {
    public override void SetData(XtraReport report, Stream stream) {
        report.SaveLayoutToXml(stream);
    }
}